contrib/database/sql/sql_test.go

```diff
@@ -268,6 +271,56 @@ func TestOpenOptions(t *testing.T) {
 		s0 := spans[0]
 		assert.Equal(t, "register-override", s0.Tag(ext.ServiceName))
 	})
+
+	t.Run("WithDBStats", func(t *testing.T) {
+		Register(driverName, &pq.Driver{})
+		defer unregister(driverName)
+		_, err := Open(driverName, dsn, WithDBStats())
+		require.NoError(t, err)
+
+		var tg statsdtest.TestStatsdClient
+		sc := internal.NewStatsCarrier(&tg)
+		sc.Start()
+		defer sc.Stop()
+		globalconfig.SetStatsCarrier(sc)
+
+		// check that within 13s, at least one round of `pollDBStats` has completed
+		deadline := time.Now().Add(13 * time.Second)
+		for {
+			if time.Now().After(deadline) {
+				t.Fatalf("Stats not collected in expected interval of %v", interval)
+			}
+			calls := tg.CallNames()
+			if len(calls) < 9 {
+				time.Sleep(100 * time.Millisecond)
+				continue
+			}
+			if containsAllStats(calls) {
+				break
+			}
+			t.Fatalf("Some stats missing")
+		}
+	})
+}
+
+func containsAllStats(calls []string) bool {
+	wantStats := []string{MaxOpenConnections, OpenConnections, InUse, Idle, WaitCount, WaitDuration, MaxIdleClosed, MaxIdleTimeClosed, MaxLifetimeClosed}
+	for _, s := range wantStats {
+		if !contains(calls, s) {
+			return false
+		}
+	}
+	return true
+}
+
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+
+	return false
```

Comment:
I recommend just replacing this with https://pkg.go.dev/github.com/stretchr/testify@v1.8.4/assert#Contains (like our other tests)

Reply:
Looks like `assert.Contains` only accepts 1 arg for the `contains` item, so I'd have to do something like:
`assert.Contains(t, s, stat1) && assert.Contains(t, s, stat2) && ... && assert.Contains(t, s, stat9)`
So I thought this way would just be cleaner but LMK what you think

Reply:
I went ahead and just made the changes I was recommending, so that we could get this merged. It wasn't a massive deal anyway 😄